File: app/src/main/java/com/usher/demo/view/looprecycler/LoopRecyclerActivity.java
package com.usher.demo.view.looprecycler;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.twigcodes.ui.layoutmanager.LoopLayoutManager;
import com.usher.demo.R;
import com.usher.demo.base.BaseActivity;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class LoopRecyclerActivity extends BaseActivity {
    @BindView(R.id.vertical_recyclerview)
    RecyclerView mVerticalRecyclerView;

    @BindView(R.id.horizontal_recyclerview)
    RecyclerView mHorizontalRecyclerView;

    private final List<String> mData = new ArrayList<>();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_loop_recycler);
        ButterKnife.bind(this);

        initData();
        initView();
    }

    private void initData() {
        for (int i = 0; i < 10; i++) {
            mData.add(String.valueOf(i));
        }
    }

    private void initView() {
        LoopLayoutManager manager = new LoopLayoutManager();

        mVerticalRecyclerView.setLayoutManager(manager);
        mVerticalRecyclerView.setAdapter(new VerticalAdapter(mData));
        mVerticalRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
                if (newState == RecyclerView.SCROLL_STATE_IDLE) {
                    int position = manager.adjustPosition();
                    Log.i("zzh", "idle " + position);
                }
            }
        });

        LoopLayoutManager manager2 = new LoopLayoutManager(LoopLayoutManager.Orientaion.HORIZONTAL);
        mHorizontalRecyclerView.setLayoutManager(manager2);
        mHorizontalRecyclerView.setAdapter(new HorizontalAdapter(mData));
        mHorizontalRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
                if (newState == RecyclerView.SCROLL_STATE_IDLE) {
                    int position = manager2.adjustPosition();
                    Log.i("zzh", "idle " + position);
                }
            }
        });

    }

    class VerticalAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

        VerticalAdapter(@Nullable List<String> data) {
            super(R.layout.item_loop_recycler_vertical, data);
        }

        @Override
        protected void convert(BaseViewHolder helper, String item) {
            helper.setText(R.id.name_textview, item);
        }
    }

    class HorizontalAdapter extends BaseQuickAdapter<String, BaseViewHolder> {

        HorizontalAdapter(@Nullable List<String> data) {
            super(R.layout.item_loop_recycler_horizontal, data);
        }

        @Override
        protected void convert(BaseViewHolder helper, String item) {
            helper.setText(R.id.name_textview, item);
        }
    }
}

File: app/src/main/java/com/usher/demo/view/LoopRecyclerActivity.kt
package com.usher.demo.view

import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.chad.library.adapter.base.BaseViewHolder
import com.jakewharton.rxbinding4.recyclerview.scrollStateChanges
import com.twigcodes.ui.adapter.RxBaseQuickAdapter
import com.twigcodes.ui.layoutmanager.LoopLayoutManager
import com.twigcodes.ui.util.RxUtil
import com.usher.demo.R
import com.usher.demo.base.BaseActivity
import kotlinx.android.synthetic.main.activity_loop_recycler.*

class LoopRecyclerActivity : BaseActivity(R.layout.activity_loop_recycler, Theme.LIGHT_AUTO) {

    override fun initView() {
        val data = List(10) { "$it" }

        vertical_recyclerview.layoutManager = LoopLayoutManager(RecyclerView.VERTICAL)
        vertical_recyclerview.adapter = VerticalAdapter(data)
        vertical_recyclerview.scrollStateChanges()
                .filter { it == RecyclerView.SCROLL_STATE_IDLE }
                .to(RxUtil.autoDispose(this))
                .subscribe {
                    (vertical_recyclerview.layoutManager as LoopLayoutManager).getFirstViewPositionWithRevision(vertical_recyclerview)
                }

        horizontal_recyclerview.layoutManager = LoopLayoutManager(RecyclerView.HORIZONTAL)
        horizontal_recyclerview.adapter = HorizontalAdapter(data)
        horizontal_recyclerview.scrollStateChanges()
                .filter { it == RecyclerView.SCROLL_STATE_IDLE }
                .to(RxUtil.autoDispose(this))
                .subscribe {
                    (horizontal_recyclerview.layoutManager as LoopLayoutManager).getFirstViewPositionWithRevision(horizontal_recyclerview)
                }
    }

    private class VerticalAdapter(data: List<String>) : RxBaseQuickAdapter<String, BaseViewHolder>(R.layout.item_loop_recycler_vertical, data) {
        override fun convert(holder: BaseViewHolder, item: String) {
            (holder.itemView as TextView).text = item
        }
    }

    private class HorizontalAdapter(data: List<String>) : RxBaseQuickAdapter<String, BaseViewHolder>(R.layout.item_loop_recycler_horizontal, data) {
        override fun convert(holder: BaseViewHolder, item: String) {
            (holder.itemView as TextView).text = item
        }
    }
}